packages/cdk/lib/stacks/engines/batch-stack.ts

```diff
@@ -56,15 +59,15 @@ export class BatchStack extends NestedStack {
     }
   }
 
-  private renderBatch(id: string, vpc: IVpc, appParams: ContextAppParameters, computeType?: ComputeResourceType): Batch {
+  private renderBatch(id: string, vpc: IVpc, appParams: ContextAppParameters, parent: Stack, computeType?: ComputeResourceType): Batch {
     return new Batch(this, id, {
       vpc,
       computeType,
       instanceTypes: appParams.instanceTypes,
       maxVCpus: appParams.maxVCpus,
       launchTemplateData: LAUNCH_TEMPLATE,
       awsPolicyNames: ["AmazonSSMManagedInstanceCore", "CloudWatchAgentServerPolicy"],
-      resourceTags: this.nestedStackParent?.tags.tagValues(),
+      resourceTags: parent.tags.tagValues(),
```

Comment:
Rather than pass around the parent stack you can use the following to look up the stack from the construct node hierarchy:

```typescript
import { Stack } from "monocdk";
// ...
const stack = Stack.of(this);
```